test/e2e/secrets_test.go

```diff
@@ -95,17 +95,17 @@ func (tc *testContext) testUserDataTamper(t *testing.T) {
 	assert.NoError(t, tc.waitForValidUserData(validUserDataSecret), "error waiting for valid userdata")
 }
 
-// waitForNewMachineNodes returns an error if waitForWindowsNodes returns the same Machine backed nodes
+// waitForNewMachineNodes returns an error if waitForConfiguredWindowsNodes returns the same Machine backed nodes
 func (tc *testContext) waitForNewMachineNodes() error {
 	var oldNodes []string
 	for _, node := range gc.machineNodes {
 		oldNodes = append(oldNodes, node.GetName())
 	}
 
-	// waitForWindowsNodes will re-populate gc.machineNodes with the configured nodes found
+	// waitForConfiguredWindowsNodes will re-populate gc.machineNodes with the configured nodes found
 	log.Printf("waiting for existing Machine nodes to be removed and replaced")
 	return wait.Poll(retryInterval, time.Minute*10, func() (done bool, err error) {
-		err = tc.waitForWindowsNodes(gc.numberOfMachineNodes, false, false, false)
+		err = tc.waitForConfiguredWindowsNodes(gc.numberOfMachineNodes, false, false)
```

Comment:
we no longer ensure the nodes become schedulable after private key change. is this intended?

Reply:
+1

Reply:
Added the check below